### PR TITLE
drivers: entropy stm32 RNG driver should not block if Seed error occurs

### LINKED ISSUE #43066: stm32wl55 true RNG  falls in seed error
**Describe the bug**
after several succesfull test, the tests/crypto/rand32/ on the nucleo_wl55jc target  fails in a infinite loop waiting for the Data Ready bit of the RNG Status reg.

Please also mention any information which could help others to understand
the problem you're facing:
 - What target platform are you using?
 - What have you tried to diagnose or workaround this issue?
 - ...

**To Reproduce**
1. west build -p auto -b nucleo_wl55jc  tests/crypto/rand32/
1bis. idem with west build -p auto -b nucleo_wl55jc -- -DOVERLAY_CONFIG=prj_hw_random_xoshiro.conf  tests/crypto/rand32/
2. west flash
3. repeat : reset board after successful execution

**Expected behavior**
```
*** Booting Zephyr OS build v3.0.0-rc3-48-gd353e5516079  ***                    
Running test suite common_test                                                  
===================================================================             
START - test_rand32                                                             
Generating random numbers                                                       
Generating bulk fill random numbers                                             
Generating bulk fill cryptographically secure random numbers                    
 PASS - test_rand32 in 0.13 seconds                                             
===================================================================             
Test suite common_test succeeded                                                
===================================================================             
PROJECT EXECUTION SUCCESSFUL   
```                                                 

**Impact**
test campaign failure

**Logs and console output**
```
*** Booting Zephyr OS build v3.0.0-rc3-48-gd353e5516079  ***                    
Running test suite common_test                                                  
===================================================================             
START - test_rand32 

```
**Environment (please complete the following information):**
 - Toolchain (e.g Zephyr SDK 0.13.2, ...)
 - Commit SHA or Version used  : 2eaede53af7afcf42dc729355ca22561922a4446

**Additional context**
in drivers/entropy/entropy_stm32.c  the `entropy_stm32_rng_get_entropy_isr()` function loops on `LL_RNG_IsActiveFlag_DRDY`
It appears that the TrueRNG RNGx->SR status register is read = 0x00000044  --> **Seed error occured**

### PR DESCRIPTION
**Pull request by FRASTM**
When a Seed error occurs during the random nb generation,
the driver tries to recover and exit without providing a random data
This avoids looping endlessly on the DRDY bit of the RNG status reg
because it remains 0 in case of error.

Fixes https://github.com/zephyrproject-rtos/zephyr/issues/43066

Signed-off-by: Francois Ramu <francois.ramu@st.com>

### CODE REVIEW

File: drivers/entropy/entropy_stm32.c [hunk 1]
```diff
@@ -361,6 +361,18 @@ static int entropy_stm32_rng_get_entropy_isr(const struct device *dev,
 		irq_disable(IRQN);
 		irq_unlock(key);
 
+		/* do not proceed if a Seed error occurred */
+		if (LL_RNG_IsActiveFlag_SECS(entropy_stm32_rng_data.rng)
+			|| LL_RNG_IsActiveFlag_SEIS(entropy_stm32_rng_data.rng)) {
```
Comment (erwango) [inline drivers/entropy/entropy_stm32.c:366]:
```suggestion
		if (LL_RNG_IsActiveFlag_SECS(entropy_stm32_rng_data.rng) ||
		    LL_RNG_IsActiveFlag_SEIS(entropy_stm32_rng_data.rng)) {
```

Comment (FRASTM) [reply drivers/entropy/entropy_stm32.c]: change line

**Review by erwango — APPROVED** [review]

File: drivers/entropy/entropy_stm32.c [hunk 2]
```diff
@@ -361,6 +361,18 @@ static int entropy_stm32_rng_get_entropy_isr(const struct device *dev,
...
+		if (LL_RNG_IsActiveFlag_SECS(entropy_stm32_rng_data.rng) ||
+			LL_RNG_IsActiveFlag_SEIS(entropy_stm32_rng_data.rng)) {
+
+			(void)random_byte_get(); /* this will recover the error */
+			/* restore irq as we enter */
+			if (irq_enabled) {
+				irq_enable(IRQN);
+			}
+			return 0; /* return cnt is null : no random data available */
+		}
```
Comment (ceolin) [inline drivers/entropy/entropy_stm32.c:374]: wouldn't be better to return some error code here instead of 0 ?

Comment (FRASTM) [reply drivers/entropy/entropy_stm32.c:374]: yes the function could  `return -EIO; ` likle in the _recover_seed_error_ function, on the other hand, the "number – of bytes filled with entropy" is actually 0.

**Comment by ceolin** [comment]
> yes the function could  `return -EIO; ` likle in the _recover_seed_error_ function, on the other hand, the "number – of bytes filled with entropy" is actually 0, in this situation. 

yes, but it was caused due a seed error and not because there is no data available. I agree that this is kinda of blurry area. It is fine for me the changes.

**Review by ceolin — APPROVED** [review]

**Pull request merged**